### DISCUSSION TITLE
How to I stop tasmota spamming MQTT stat topic with rule commands results?

### DISCUSSION THREAD

**Discussion by Vitek-117** [arendst/Tasmota#21740] [Q&A] [answered]
I have a rule that runs every second, assigning a value to a variable
But tasmota posts the “result” every time it’s calculated, to stat/devicename/RESULT topic
Despite the fact that the Telemetry Sending Period is specified in the settings for 90 seconds.
![Screenshot_3](https://github.com/arendst/Tasmota/assets/133482412/d7a0dbda-26e1-4501-af03-f20f13006b50)
How can I STOP it doing that? 

My rule is Rule1 ON DS18B20#Temperature DO DisplayText %value%^ ENDON

Thanks

**Comment by sfromis** [comment]
You can't. Just ignore what you don't need. Still, you could reduce the amount by amending the rule to only update the display if the temperature value changes.

In addition, there is a feature for displaying data from JSON without rules....
https://tasmota.github.io/docs/Displays/#display-json-variables

**Reply by Vitek-117** [reply]
I have a TM1637 display. How can I change the rule to update the display only when the temperature value changes? I found something similar: Rule2 on DS18B20#Temperature!=%var1% do backlog displaytext %value%^;var1 %value% endon 
It's a little better than it was. But it's still not what you need. Because a tenth of a degree is constantly floating, and mqtt continues to spam.
Of course, you can use the rule: 
ON Tele-DS18B20#Temperature DO DisplayText %value%^ ENDON 
But then I won't see the temperature change in real time on the display. But I don't want mqtt to spam the results from the display rule. And for sending to be set as in the settings, the period for sending telemetry is 90 seconds.
![Screenshot_2](https://github.com/arendst/Tasmota/assets/133482412/d6dac96c-11e1-482f-ab84-3791c87bb18e)

**Reply by sfromis** [reply]
Your main issue seems to be thinking of it as something which is a problem, "spam". A **tiny** bit of network activity is only an issue if you sit and stare at it :grinning:

**Reply by sfromis** [reply]
A very new feature of Tasmota with latest development builds is that if using `Backlog3` (or `Backlog2` without the inter-command delay), Tasmota will no longer report the command execution via MQTT. That also goes if you prefix an individual command name with underscore.

**Reply by Vitek-117** [reply]
I apologize for the long answer.  
Backlog2 "on DS18B20#Temperature DO DisplayText %value%^endon"

Did I write the command correctly? Will it work?

**Reply by sfromis** [reply]
No, using `Backlog3` (with just one command, `Backlog2` provides no benefit) is on the command you want no MQTT result from, in this case the `DisplayText` command. And those `"` are not part of the syntax. Also, spaces the right places are important, also before `endon`.

**Reply by Vitek-117** [reply]
Backlog3 on DS18B20#Temperature DO DisplayText %value%^ endon

Is that right now? If I made a mistake in writing the command again, then please write the correct command. So that I don't bother you with unnecessary questions. What is the difference between backlog 2 and backlog 3?

**Reply by Vitek-117** [reply]
Where can I read about backlog2 and backlog 3? I didn't find anything by searching.

**Reply by barbudor** [reply]
Backlog3 is like Backlog but prevent sending the result of the command(s)
Backlog2 is like Backlog0 but prevent sending the result of the command(s)

You need to use Backlog in the DO part of the rule:
```
ON trigger DO Backlog command1; command2; command 3 ENDON
```

**Reply by Vitek-117** [reply]
[barbudor]
Rule2 ON DS18B20#Temperature DO Backlog3 DisplayText %value%^ ENDON
Did I understand you correctly and wrote the rule?

**Reply by Vitek-117** [reply]
[barbudor]
Backlog3 is like Backlog but prevent sending the result of the command(s)
Backlog2 is like Backlog0 but prevent sending the result of the command(s)
You wrote the same thing. I still don't understand the difference...

**Reply by barbudor** [reply]
Yes the syntax is correct

No I didn't wrote the same thing as there is a différence (documented) between Backlog and Backlog0, the same différence applies between Backlog3 and Backlog2 when there are multiple subcommands in backlog

But as pointed by sfromis, as you only have 1 command in your backlog, Backlog3 or Backlog2 will behave the same.

**Reply by Vitek-117** [reply]
Please write the full command. So I just added it and it worked. Because I can't figure out how to do it right.  Since I am not a professional programmer. What I wrote doesn't work. Where else should I add backlog3 to make it work?
Backlog3 ON DS18B20#Temperature DO Backlog3 DisplayText %value%^ ENDON
Right?

**Reply by sfromis** [reply]
No, you should *only* use Backlog3 for the DisplayText command, not when entering a rule.

And since you originally posted rules without including the `Rule` command, it appeared that you understood that you always has to input rule sets via that command.

As barbudor confirmed, you already had the correct syntax:
`Rule2 ON DS18B20#Temperature DO Backlog3 DisplayText %value%^ ENDON`

To be sure, I also posted that you have to use most recent development build of Tasmota, not a release version, as it is a very new feature to exclude MQTT publish for command results.

**Reply by Vitek-117** [reply]
[sfromis]
I understand correctly that my team(Rule2 ON DS18B20#Temperature DO Backlog3 DisplayText %value%^ ENDON) will ONLY work with the latest version of beta. And it won'T work with the latest release. I have Tasmota version 14.1.0.3 
Am I right?
I do not insist on using a rule to execute this command. If possible, embed it in this line: 
define USER_BACKLOG "template  {\"NAME\":\"MY_WT32-ETH01_Display_switch\",\"GPIO\":[0,0,7136,0,7104,1312,0,0,224,0,225,226,3840,576,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,608,640,0,160,161,0,0,162],\"FLAG\":0,\"BASE\":5}; Teleperiod; wifi 0; Rule1 ON; Rule2 ON; EthIpAddress 192.168.1.54; EthGateway 192.168.1.254; EthSubnetmask 255.255.255.0; EthDNSServer 8.8.8.8; SwitchMode 8; SetOption146 1; DisplayMode 0; DisplayModel 15; module 0; "
Then it will be even better and easier.

**Reply by sfromis** [reply]
Instead of having it all in `USER_BACKLOG`, I'd recommend following the intended structure of having the rule set(s) in `USER_RULE1` / `USER_RULE2` / `USER_RULE3`, and template in `USER_TEMPLATE`. Then a `#define MODULE USER_MODULE` is the direct way of having the template enabled too, without a command.

**Reply by Vitek-117** [reply]
Understood you. Then I will use USER_RULE 2
Please provide a link to the latest version of tasmota. Where there is a new feature to exclude MQTT publishing for team results.

**Reply by sfromis** [reply]
No, `USER_RULE 2` will not work, spaces are important, it should be `USER_RULE2`

And it is just standard Tasmota source as long as it was downloaded within the last 5 days.

**Reply by Vitek-117** [reply]
Yes, I know that there should be no space in USER_RULE2. It's because of the translator. It automatically puts spaces where it is not necessary. Sorry for the inaccuracy.
I'll try to download the latest source code. Since the one I use, I downloaded it more than a week ago.

**Comment by uiolee** [comment]
same question. If I have multiple devices like this, I'm not sure if my fragile mqtt broker can handle it.

**Reply by barbudor** [reply]
1) answers have been provided above. If your problem is the same, then same solution applies. If it's not the same, don't hijack this topic and open a new one

2) if your broker is weak, start by getting something correct. Mosquitto is free and rock solid

**Reply by sfromis** [reply]
With a mere "multiple" you should generally not worry about overloading a broker. Now, if "multiple" mean thousands of devices, you may have to make sure to have a robust structure. MQTT is a very lightweight protocol, and the amount of MQTT traffic shown above is **tiny** in the context of a network.

**Comment by uiolee** [comment]
Thank you, this is useful to me. 
And it seems it hasn't been merged into the master branch yet

https://github.com/arendst/Tasmota/blob/990a0e4257b9d180d1ad70cd8da71ddaef7f91b1/tasmota/tasmota_support/support_command.ino#L494-L495

**Reply by sfromis** [reply]
That's why I twice reminded about needing to use latest development build.

**Comment by Vitek-117** [comment, marked as answer]
The issue with the "spam" from MQTT has been resolved. What should be done to avoid spam: 
1. Downloaded the latest version of the original Tasmota version 14.1.0.4, and compiled it to suit your needs.
2. Added the underscore character "_" before the _DisplayText command.
The full command looks like this: USER_RULE2 "ON DS18B20#Temperature DO _DisplayText %value%^ ENDON" 
After that, the MQTT spam stopped.

![Screenshot_1](https://github.com/user-attachments/assets/86e9a396-94a2-44a5-ba49-1304f9560428)

**Comment by Vitek-117** [comment]
Thank you all! The problem is solved.

**Comment by phantu91git** [comment]
i have same question. using tasmota 14.1.0.4. here is mqtt explorer log. it increase dramatically about at least 5 message in one sec. i want to reduce the write cylcle in broker sd card so how to use the rule to report only 25 second.
![image](https://github.com/user-attachments/assets/f6f8e4e5-10f2-4206-9f7f-439914879336)
```
}

00:12:21.942 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:21","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[69.0,30.0,34.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:22.941 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:22","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[66.0,55.0,33.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:24.941 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:24","BMP280":{"Temperature":31.1,"Pressure":1006.2},"LD2410":{"Distance":[63.0,75.0,36.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:25.942 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:25","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[39.0,51.0,39.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:26.942 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:26","BMP280":{"Temperature":31.1,"Pressure":1006.2},"LD2410":{"Distance":[30.0,30.0,0.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:29.942 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:29","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[30.0,30.0,0.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:30.942 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:30","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[60.0,67.0,0.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:31.941 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:31","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[30.0,54.0,34.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:32.942 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:32","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[56.0,30.0,0.0],"Energy":[96,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:34.942 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:34","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[72.0,30.0,0.0],"Energy":[91,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:35.941 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:35","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[30.0,30.0,0.0],"Energy":[74,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:38.938 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:38","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[30.0,30.0,66.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:39.939 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:39","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[30.0,30.0,0.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:40.938 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:40","BMP280":{"Temperature":31.1,"Pressure":1006.2},"LD2410":{"Distance":[41.0,61.0,0.0],"Energy":[93,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:41.939 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:41","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[30.0,30.0,32.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:44.946 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:44","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[48.0,44.0,0.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:45.027 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:45","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[48.0,44.0,0.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:45.938 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:45","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[30.0,63.0,0.0],"Energy":[71,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:50.938 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:50","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[30.0,30.0,0.0],"Energy":[96,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:52.939 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:52","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[230.0,71.0,50.0],"Energy":[31,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:53.937 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:53","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[66.0,72.0,59.0],"Energy":[94,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:55.010 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:55","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[58.0,60.0,144.0],"Energy":[69,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:55.937 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:55","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[59.0,58.0,69.0],"Energy":[76,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:12:58.941 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:12:58","BMP280":{"Temperature":31.1,"Pressure":1006.2},"LD2410":{"Distance":[74.0,71.0,79.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:13:01.940 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:13:01","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[69.0,61.0,46.0],"Energy":[67,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:13:02.941 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:13:02","BMP280":{"Temperature":31.1,"Pressure":1006.1},"LD2410":{"Distance":[62.0,67.0,46.0],"Energy":[100,100]},"PressureUnit":"hPa","TempUnit":"C"}
00:13:04.941 MQT: tasmota/T3_P2/cambien/khac/_00/SENSOR = {"Time":"1970-01-01T00:13:04","BMP280":{"Temperature":31.1,"Pressure":1006.2},"LD2410":{"Distance":[64.0,55.0,42.0],"Energy":[76,100]},"PressureUnit":"hPa","TempUnit":"C"}

```
![image](https://github.com/user-attachments/assets/f644a4bb-d25f-4259-8fe5-a21eaf53cfb5)

**Reply by barbudor** [reply]
Your configuration looks weird
1. You don't have a proper time set. Tasmota requires a proper time, so fix your NTP configuration to get your device on time
2. You tweaked the fulltopic beyond reasonable as the prefix is no more there. So tasmota has no more way to distinguish between commands, responses and telemetry messages

Fix those first

**Reply by barbudor** [reply]
And your problem is not related to that thread as those posts don't Come from a rule but from obvious misconfiguration

**Reply by phantu91git** [reply]
> Your configuration looks weird
> 
>     1. You don't have a proper time set. Tasmota requires a proper time, so fix your NTP configuration to get your device on time
> 
>     2. You tweaked the fulltopic beyond reasonable as the prefix is no more there. So tasmota has no more way to distinguish between commands, responses and telemetry messages
> 
> 
> Fix those first

thank for your time. but i think i not mess up too much with configuration.
1. that router have some tuya device too connected to it so i decide to isolate it from main-network as well as internet for security concern. but even that i think timer on message should be control by software not hardware.
2.  the full-topic which i inheritance from zigbee2mqtt topic style as well as easy_to_me_to_manage home block. even i revert back to default config :
![image](https://github.com/user-attachments/assets/c5dcd6f9-c346-4be0-be85-4d84ab2b0f82)
it should be the same for mqtt explorer: for quite some second there are over 250 message :
![image](https://github.com/user-attachments/assets/973cc643-d97c-4a5a-9aa0-656d57af2c98)
 when the sensor ld2410 trigger. i wonder how to limit the mqtt rate about (5 sec or 10 sec) to report back using the Rule.
this statement "tasmota has no more way to distinguish between commands, responses and telemetry messages" so which configuration in the firmware decide "responses and telemetry message", i think i just push only neccessary stuff on "user_config_override.h", so can you clarify to me because i look for documentation but mostly it is inside the documentation recommend
![image](https://github.com/user-attachments/assets/4f5eec1e-f3b9-4369-8db2-249ee41e3908)

i reopen the router to internet, the problems till persist
![image](https://github.com/user-attachments/assets/d2be52db-a00b-4c36-aa22-d6d511fdaa4f)
funny thing is it getting quite much message, more  than 8 zigbee2mqtt device
![image](https://github.com/user-attachments/assets/ace60a17-6ea2-4122-b3cb-8a470f81efd6)

**Reply by sfromis** [reply]
With the LD2410, you *will* get activity data when there's movement, as the whole purpose of such a sensor is to follow activity. And yes, as this is not related to "rule command results", this is indeed not really on-topic for this thread.

**Reply by phantu91git** [reply]
> With the LD2410, you _will_ get activity data when there's movement, as the whole purpose of such a sensor is to follow activity. As this is not related to "rule command results", this is not really on-topic for this thread.

so how do i limit the mqtt publish rate, i don't need too accurate timing. so how to pubish only 1 message per 5 second or 10 second. i am searching for command documentation, is it a way to custom the command to public the mqtt message from sensor?

**Reply by sfromis** [reply]
I just said that "you *will* get activity data when there's movement", meaning that there is no feature for not getting it. While you can have a command doing *additional* publish, this will not affect the standard publishing done by the driver. Anyway, about one message a second is nothing in MQTT, and using MQTT should not cause the data to be written to your SD card (not a great storage medium for a server).

And it is not a command doing the publish, it is the driver for the sensor.

**Reply by phantu91git** [reply]
> I just said that "you _will_ get activity data when there's movement", meaning that there is no feature for not getting it. While you can have a command doing _additional_ publish, this will not affect the standard publishing done by the driver. Anyway, about one message a second is nothing in MQTT, and using MQTT should not cause the data to be written to your SD card (not a great storage medium for a server).
> 
> And it is not a command doing the publish, it is the driver for the sensor.

i figured it out about that. it should take time to look at the core of tasmota. however i wonder about mqtt standard publishing done by the driver. is it internal "rule" to tasmota to manual publishing it or it tied to the mqtt web page configuration and it internal procedure. i understand that u guys do the great job, but the reason is that i concern about quite a lof of thing with "radio signal output devices", my own opinion is " the lesser the better for health". and of course, it is not good for using sd card but the setup for sd for single board computer in my opinion is much better for maintain as well as introduce to other in case of need.

**Reply by Jason2866** [reply]
Publishing a mqtt telegram is initiated from the related driver. Thats it. What do you mean with mqtt Web page? There is no such thing.

**Reply by sfromis** [reply]
As I said, the driver is doing the publish when it has detected movement. Nothing to be added there. This is not some sort of "manual" (not sure what you mean by that), or related to configuration.

If you feel a need to worry about health and minuscule RF sources, you should remove all electricity from your home, as you get vastly more RF from just normal household things. Of course, you can still get a lot of "ambient" RF in regular residential areas. Even the LD2410 itself is *constantly* emitting more RF activity than resulting from the MQTT publish.

And there is nothing about using MQTT which will cause SD card writes depending on activity, unless you make the choice of writing stuff to the card every time you get something.

**Reply by phantu91git** [reply]
> Publishing a mqtt telegram is initiated from the related driver. Thats it. What do you mean with mqtt Web page? There is no such thing.

i mean something like this. i found this one on the ld2410_code
```
void Ld2410EverySecond(void) {
  if (LD2410.moving_energy) {
    // Send state change to be captured by rules
    // {"Time":"2022-11-26T10:48:16","Switch1":"ON","LD2410":{"Distance":[125.0,0.0,0.0],"Energy":[0,100]}}
    MqttPublishSensor();
  }
}
```
i mean "the manual" can be putting a wrapping function " a debounce similar function to wait for specific time like 5 sec before sending, normally understand how the core work is better to custom thing", in my point of view, i think the way tasmota work is "send that driver thought mqtt".when i mean the configuration mqtt web page, how it tie to the way mqtt message to be send. but based on my understand, the web page just a configuration for mqtt host-client.

**Reply by phantu91git** [reply]
> As I said, the driver is doing the publish when it has detected movement. Nothing to be added there. This is not some sort of "manual" (not sure what you mean by that), or related to configuration.
> 
> If you feel a need to worry about health and minuscule RF sources, you should remove all electricity from your home, as you get vastly more RF from just normal household things. Of course, you can still get a lot of "ambient" RF in regular residential areas. Even the LD2410 itself is _constantly_ emitting more RF activity than resulting from the MQTT publish.
> 
> And there is nothing about using MQTT which will cause SD card writes depending on activity, unless you make the choice of writing stuff to the card every time you get something.

i understood that. i have a bed room for kid with faraday grounding and of course limit my kids from interact with electricity devices. thank for these other clarify.

**Reply by Jason2866** [reply]
Fork and do whatever you want in the code. Tasmota is Open Source.

**Reply by sfromis** [reply]
Only publishing after some interval will destroy the function of having the sensor, unless you accumulate change actively for the measurements during the period. But still, the RF from using the LD2410 is vastly more than from the Wifi activity for publishing data.
